Add edit clinic series post (#523)
NHSDigital/prevention-services-design-history: app/manage-your-appointments/2026/06/making-it-easier-to-manage-clinics/index.md

---
title: "Making it easier to manage clinics"
author: Joe Julier
date: 2026-06-01
tags:
  - pharmacies
  - appointments
---

> In [a previous post](/manage-your-appointments/2026/01/helping-users-cancel-or-edit-multiple-sessions) we documented some of our early attempts to make it easier to edit multiple sessions in Manage Your Appointments (MYA). 

Reviewer: I think making the link "some of our early attempts to make it easier to edit multiple sessions in Manage Your Appointments (MYA)" would be more helpful to screen reader users.

> This post picks up the story now the design work is 'done' and the jira tickets are wending their way across the board on their journey into prod. 

Reviewer: 'prod' is a bit jargony?

Reviewer: Also maybe add a single sentence of what that change or solution was so the reader doesn't need to go back and read the previous post.

> This post picks up the story now the design work is 'done' and the jira tickets are wending their way across the board on their journey into prod. 

Reviewer: ```suggestion
This post picks up the story now the design work is 'done' and the Jira tickets are wending their way across the board on their journey into prod. 
```

> The problem we're solving remains the same, but solution has changed significantly. 

Reviewer: ```suggestion
The problem we're solving remains the same, but the solution has changed significantly. 
```

> This was made possible by the decision to implement recurrence rules in MYA. Choosing this approach, which stores the set of repeating clinics users create as a single editable thing, was a big call. But in the usability testing we've run the benefit for users is clear.

Reviewer: The sentence is hard to parse, it hides who did what. Something like "This works because the team built recurrence rules into MYA." Would help me.

> This was made possible by the decision to implement recurrence rules in MYA. Choosing this approach, which stores the set of repeating clinics users create as a single editable thing, was a big call. But in the usability testing we've run the benefit for users is clear.

Reviewer: ```suggestion
This was made possible by the decision to implement recurrence rules in MYA. Choosing this approach, which stores the set of repeating clinics users create as a single editable thing, was a big call. But in the usability testing we've run, the benefit for users is clear.
```

While we've got ambitions to move towards a calendar style interface, that wasn't deliverable within the time constraints we're working to. But we thought it made sense to borrow from their language so decided to refer to a repeating set of clinics as a 'clinic series'. 

In usability testing this worked well, without being perfect. Users understood it, but also said it didn't match the real world language they use. However, when we assessed the other options available we found clinic series was the clearest option for key parts of the service, such as confirmation pages.

## New navigation

### Before

![Screenshot of MYA navigation showing separate top-level sections for “View appointments” and “Manage clinics”, with sub-navigation for Day, Week and Month views.](old-nav.png '')

Previously there were two areas in the top-level navigation: 'Create Availability' and 'View Availability'. 

> 'Create Availability' was purely for creating clinics and viewing the confusing table of audit events. If you wanted to change those clinics you had to go to 'View Availability', navigate to the week, and select the specific clinic that you wanted to change.

Reviewer: Started using title case for nav labels. Think its sentence case?

### Iteration 1

> ![Screenshot of MYA navigation showing separate top-level sections for “View appointments” and “Manage clinics”, with sub-navigation for Day, Week and Month views.](nav-iteration-1.png '')

Reviewer: This alt text is same as previous image, I think its wrong

In our new approach, where you can edit and delete clinic series, 'Create Availability' was no longer accurate. 

> We tried replacing is with 'Manage clinics' to capture the wider range of actions now possible. But in our first round of usability testing this resulted in users expecting all clinic management to live in this section. Whereas to change individual clinics users still needed to go to the week view within 'View apppointments'.

Reviewer: ```suggestion
We tried replacing it with 'Manage clinics' to capture the wider range of actions now possible. But in our first round of usability testing this resulted in users expecting all clinic management to live in this section. Whereas to change individual clinics users still needed to go to the week view within 'View appointments'.
```

### Iteration 2

![Screenshot of MYA navigation showing a single “Clinics” section with sub-navigation options: Day view, Week view, Month view, and “Clinics list”](nav-iteration-2.png '')

In our second iteration we combined the two sections into a single area called Clinics, and what was previously 'Manage clinics' became 'Clinics list' in the sub-navigation. 

> Putting all four views of clinics (from a single day up to a summary of everything) in the same sub-navigation made it easier for users to switch between the them. In the second round of testing that people had far less trouble changing individual clinics. 

Reviewer: ```suggestion
Putting all four views of clinics (from a single day up to a summary of everything) in the same sub-navigation made it easier for users to switch between them. In the second round of testing we found that people had far less trouble changing individual clinics. 
```

![Diagram showing an overview screen leading to separate, linear edit journeys. Each change is made in its own flow, with users completing and confirming one change before returning to the overview to start another](edit-clinics-2.png '')

In the vast majority of cases users only want to make one change at a time, so in our second iteration we treated the overview screen purely as the entry point into several linear edit journeys.

## Handling conflicts

Introducing the ability to edit a whole clinic series also meant figuring out how to handle the interplay between changes to the whole series, and individual clinics within it. 

Rather than forcing users to decide what they want to do whenever there is a conflict, we worked with the team to come up with some default rules which are applied automatically:

* Start and end times - edits to individual clinics are preserved
* Number of vaccinators - edits to individual clinics are preserved
* Services - edits to individual clinics are preserved but if new services are added to the series these are also applied to the individual clinic
* Closures - changes to the series are applied regardless of existing edits to individual clinics

> These rules were based on our understanding of how our users work, but we still sense checked them in research sessions. We found they aligned with our users expectations and needs. 

Reviewer: ```suggestion
These rules were based on our understanding of how our users work, but we still sense checked them in research sessions. We found they aligned with our users’ expectations and needs. 
```

In our first iteration we tried to communicate a conflict between a change to a clinic series and the individual clinics within it via a screen in the change series flow. 

Although participants did manage (with some concerted effort) to understand what was happening when asked to revisit this screen, almost no one grasped it when in task-completion mode. The message was perceived as a confusing distraction. 

### First release

In our second iteration we showed individual clinics in the series with edits in the series overview screen:

![Screenshot of a clinic series overview showing closures and individual clinics with changes. The page lists closure dates and highlights a specific clinic with edited times, with options to change each section.](affected-clinics-1.png '')

We also listed any clinics that hadn't been updated due to a conflict on the success screen:

![Screenshot of a confirmation page showing a clinic series has been updated, with a message explaining that 2 clinics were not updated due to previous changes. The page lists the affected clinic dates and provides options to make another change or return to the clinics list](affected-clinics-2.png '')

> This approach was much more successful. Participants grasped why some sessions hadn't been updated and were able to confirm their interpretation by viewing the clinics with changes listed in the clinic series overview.

Reviewer: Should this be 'clinics' not 'sessions'?
